### PR TITLE
feat: handle test cases with multiple types

### LINKED ISSUE #87: Handling a single test with multiple types
The following test suite from the latest N3 spec contains some tests that are *both* negative turtle tests and positive n3 tests; I'm wondering you think is the best behavior to handle these tests @rubensworks. My view is that there should be a separate test entry created *for each type* so that only one handler is needed per entry still; and then we should add a filter option to allow for whitelisting or blacklisting test types when running the spec test.

https://w3c.github.io/N3/tests/TurtleTests/manifest.ttl

### PR DESCRIPTION
**Pull request by jeswr**
Works as expected on N3.js and Comunica - and can confirm it resolves #87 .

### CODE REVIEW

**Review by rubensworks — CHANGES_REQUESTED** [review]
Thanks, code looks perfect!

Could we also add a unit test to check for the specific case of a test case with multiple types?

**Comment by jeswr** [comment]
> Could we also add a unit test to check for the specific case of a test case with multiple types?

Done!

I've also added the functionality to forward data from the test cases to the parser; in this case the main goal was to expose the expected `mediaType` of a test case (which makes the dual N3 & turtle test cases easier).

File: lib/testcase/TestCaseMediaTypes.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  'http://www.w3.org/ns/rdftest#TestTrigPositiveSyntax': "application/trig",
+  'http://www.w3.org/ns/rdftest#TestTrigNegativeSyntax': "application/trig",
+  'http://www.w3.org/ns/rdftest#TestTrigNegativeEval': "application/trig",
+
+  // JSON-LD test suite
+  'https://w3c.github.io/json-ld-api/tests/vocab#ToRDFTest https://w3c.github.io/json-ld-api/tests/vocab#PositiveEvaluationTest': "application/ld+json",
+  'https://w3c.github.io/json-ld-api/tests/vocab#ToRDFTest https://w3c.github.io/json-ld-api/tests/vocab#PositiveSyntaxTest': "application/ld+json",
+  'https://w3c.github.io/json-ld-api/tests/vocab#ToRDFTest https://w3c.github.io/json-ld-api/tests/vocab#NegativeEvaluationTest': "application/ld+json",
+  'https://w3c.github.io/json-ld-api/tests/vocab#FromRDFTest https://w3c.github.io/json-ld-api/tests/vocab#PositiveEvaluationTest': "application/ld+json",
+  'https://w3c.github.io/json-ld-api/tests/vocab#FromRDFTest https://w3c.github.io/json-ld-api/tests/vocab#NegativeEvaluationTest': "application/ld+json",
```
Comment (jeswr) [inline lib/testcase/TestCaseMediaTypes.ts:34]: This is the only thing I am now unsure about - it arises from the keys in the `TestCaseHandlers.ts` file.

Comment (rubensworks) [reply lib/testcase/TestCaseMediaTypes.ts:34]: I'm not sure if this approach can capture the media types for different test types.

For example, in the JSON-LD test suite, some tests will have an HTML content type, which depends on the presence of another test type (for example https://w3c.github.io/json-ld-api/tests/html-manifest.html#tex01).
For these tests, the test suite will even pass a custom `contentType` parameter to the parser.

Would it instead make sense to only capture the edge-cases in this file (Turtle/N3), and not do anything in all other cases?

Comment (rubensworks) [reply lib/testcase/TestCaseMediaTypes.ts:34]: Ok!

Comment (jeswr) [reply lib/testcase/TestCaseMediaTypes.ts:34]: > For example, in the JSON-LD test suite, some tests will have an HTML content type, which depends on the presence of another test type (for example https://w3c.github.io/json-ld-api/tests/html-manifest.html#tex01).
For these tests, the test suite will even pass a custom contentType parameter to the parser.


... I've also just discovered that the current implementation causes no spec tests to be run when testing against the jsonld-streaming-parser so we need to fix this as well. Debugging now.
